File: app/src/main/java/skarlat/dev/ecoproject/Authenticator.java
package skarlat.dev.ecoproject;

public abstract class Authenticator<T> {
	protected T instance;
	
	public Authenticator(T instance) {
		this.instance = instance;
	}
	
	public abstract User getCurrentUser();
	
	T getInstance(){
		return instance;
	}
}

File: app/src/main/java/skarlat/dev/ecoproject/authentication/Authenticator.kt
package skarlat.dev.ecoproject.authentication

import android.os.Bundle
import skarlat.dev.ecoproject.authentication.strategy.AuthStrategy

abstract class Authenticator {
    var authStrategy: AuthStrategy? = null
    abstract fun authenticate(bundle: Bundle)
    abstract suspend fun logout()
    abstract suspend fun register(bundle: Bundle)
}